checkov/terraform/checks/graph_checks/aws/KmsKeyWildcardPrincipal.yaml

```diff
@@ -0,0 +1,28 @@
+---
+metadata:
+  id: "CKV2_AWS_67"
+  name: "Ensure KMS key policy does not contain wildcard (*) principal"
+  category: "GENERAL_SECURITY"
+definition:
+  and:
+    - cond_type: "connection"
+      resource_types:
+            - "aws_kms_key"
+      connected_resource_types:
+            - "aws_kms_key_policy"
+      operator: "exists"
+    - cond_type: "filter"
+      attribute: "resource_type"
+      operator: "within"
+      value: "aws_kms_key"
+    - cond_type: "attribute"
+      resource_types:
+            - "aws_kms_key_policy"
+      attribute: "policy.Statement[?(@.Effect == 'Allow' & @.Principal == '*')]"
+      operator: "jsonpath_not_exists"
+    - cond_type: "attribute"
+      resource_types:
+            - "aws_kms_key_policy"
+      attribute: "policy.Statement[?(@.Effect == 'Allow')].Principal.AWS[*]"
+      operator: "jsonpath_not_equals"
+      value: "*"
```

Comment:
your policy is handling key to policy connection, but not when the policy is inlined, that's why the actual test fails for all resources.